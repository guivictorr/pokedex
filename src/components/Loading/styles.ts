import styled, { keyframes } from 'styled-components';

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  img {
    animation: ${rotate} 2s linear infinite;
  }
`;
