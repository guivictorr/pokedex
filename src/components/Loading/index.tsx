import Image from 'next/image';
import Heading from '../Heading/styles';
import * as L from './styles';

const Loading = () => (
  <L.Container>
    <Image alt="Pokebola" src="/pokeball.svg" width={100} height={100} />
    <Heading level={2} fontSize="xlarge">
      Carregando
    </Heading>
  </L.Container>
);

export default Loading;
