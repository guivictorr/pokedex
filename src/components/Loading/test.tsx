import { render, screen } from '@testing-library/react';
import Loading from '.';

describe('<Loading />', () => {
  it('should render correctly', () => {
    const { container } = render(<Loading />);

    expect(container).toMatchSnapshot();
  });

  it('should render loading message', () => {
    render(<Loading />);

    const text = screen.getByRole('heading', {
      name: /carregando/i,
    });

    expect(text).toBeInTheDocument();
  });

  it('should render pokeball image', () => {
    render(<Loading />);

    const image = screen.getByRole('img', { name: /pokebola/i });
    expect(image).toBeInTheDocument();
  });
});
