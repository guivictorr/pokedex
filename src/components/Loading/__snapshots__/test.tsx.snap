// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render correctly 1`] = `
.c1 {
  color: #1e1e1f;
  font-weight: 700;
  font-size: 36px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 img {
  -webkit-animation: eoUyJr 2s linear infinite;
  animation: eoUyJr 2s linear infinite;
}

<div>
  <div
    class="c0"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <noscript />
      <img
        alt="Pokebola"
        data-nimg="true"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <h2
      class="c1"
      font-size="xlarge"
    >
      Carregando
    </h2>
  </div>
</div>
`;
